pkg/reconciler/route/resources/certificate.go

```diff
@@ -34,7 +34,7 @@ import (
 // MakeCertificate creates a Certificate, inheriting the certClass
 // annotations from the owner, as well as the namespaces. If owner
 // does not have a certClass, use the provided `certClass` parameter.
-func MakeCertificate(owner kmeta.OwnerRefableAccessor, ownerLabelKey string, dnsName string, certName string, certClass string) *networkingv1alpha1.Certificate {
+func MakeCertificate(owner kmeta.OwnerRefableAccessor, ownerLabelKey string, dnsName string, certName string, certClass string, domain string) *networkingv1alpha1.Certificate {
```

Comment:
hmmm okay. This would a separate PR that could come after this one?

And are you imagining like a custom type for each string entry? Or that the MakeCertificate function would accept an object (perhaps even a CertificateSpec) instead of all those strings?